apps/mobile/src/features/send-refactor/forms/stx/stx-form-schema.ts

```diff
@@ -0,0 +1,118 @@
+import { useMemo } from 'react';
+
+import { errorMessages } from '@/features/send-refactor/error-messages';
+import { CalculateStxMaxSpend } from '@/features/send-refactor/hooks/use-calculate-stx-max-spend';
+import { addressComplianceValidator } from '@/features/send-refactor/utils';
+import { ChainId } from '@stacks/network';
+import * as z from 'zod';
+
+import { STX_DECIMALS } from '@leather.io/constants';
+import {
+  isValidAddressChain,
+  isValidStacksAddress,
+  isValidStacksMemo,
+  validatePayerNotRecipient,
+} from '@leather.io/stacks';
+import { isValidPrecision, stxToMicroStx } from '@leather.io/utils';
+
+interface SchemaCreationParams {
+  calculateStxMaxSpend: CalculateStxMaxSpend;
+  payerAddress: string;
+  chainId: number;
+}
+
+export function useStxSendFormSchema({
+  calculateStxMaxSpend,
+  payerAddress,
+  chainId,
+}: SchemaCreationParams) {
+  return useMemo(
+    () => createStxSendFormSchema({ calculateStxMaxSpend, payerAddress, chainId }),
+    [calculateStxMaxSpend, payerAddress, chainId]
+  );
+}
+
+function createStxSendFormSchema({
+  calculateStxMaxSpend,
+  payerAddress,
+  chainId,
+}: SchemaCreationParams) {
+  return z.object({
+    amount: z
+      .string()
+      .superRefine((amount, context) => stxAmountValidator(amount, context, calculateStxMaxSpend)),
+    recipient: z
+      .string()
+      .refine(isValidStacksAddress, errorMessages.invalidAddress)
+      .refine(stxAddressNetworkValidator(chainId), errorMessages.incorrectNetworkAddress)
+      .refine(stxNotCurrentAddressValidator(payerAddress), errorMessages.recipientIsPayer)
+      .refine(stxComplianceValidator(chainId), errorMessages.nonCompliantAddress),
+
+    memo: z.string().refine(isValidStacksMemo, errorMessages.memoExceedsLimit),
+    nonce: z.number(),
+    fee: z.number(),
+    isSendingMax: z.boolean(),
+  });
+}
+
+function stxAddressNetworkValidator(chainId: ChainId) {
```

Comment:
It could be good to move these validators into `@leather.io/stacks`? We can do it in a future task. 

That would make it easier to share this with the `extension` and replace `yup` with `zod`